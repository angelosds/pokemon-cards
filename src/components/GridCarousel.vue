<template>
  <div class="grid-carousel">
    <slot />
  </div>
</template>

<style lang="scss" scoped>
.grid-carousel {
  display: grid;
  grid-auto-flow: column;
  grid-gap: 24px;
  overflow-x: auto;
  padding: 24px;
  scroll-snap-type: x mandatory;

  > .grid-carousel--item {
    min-width: calc(100vw - 80px);
    scroll-snap-align: center;
    justify-self: center;
  }

  @media screen and (min-width: $breakpoint-mobile) {
    grid-auto-flow: row;
    grid-template-columns: repeat(auto-fit, minmax(260px, 1fr));
    overflow-x: unset;

    > .grid-carousel--item {
      min-width: auto;
    }
  }
}
</style>
