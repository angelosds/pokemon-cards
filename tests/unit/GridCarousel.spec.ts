import { shallowMount } from "@vue/test-utils";

import GridCarousel from "@/components/GridCarousel.vue";

describe("GridCarousel.vue", () => {
  const build = (slots?: object) => {
    const wrapper = shallowMount(GridCarousel, {
      slots: { ...slots }
    });

    return { wrapper };
  };

  it("renders the component", () => {
    const { wrapper } = build();

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("renders content in a default slot", () => {
    const defaultSlotText = "default";
    const { wrapper } = build({ default: defaultSlotText });

    expect(wrapper.text()).toMatch(defaultSlotText);
  });
});
