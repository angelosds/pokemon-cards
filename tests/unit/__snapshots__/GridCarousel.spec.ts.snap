// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridCarousel.vue renders the component 1`] = `<div class="grid-carousel"></div>`;
